extensions/replication/tasks/ReplicateObject.js

```diff
@@ -696,11 +711,38 @@ class ReplicateObject extends BackbeatTask {
     }
 
     _setupDestClients(targetRole, log) {
+        this.destBackbeatHost = this.destHosts.pickHost();
+
+        if (this.destConfig.auth.type === authTypeAssumeRole) {
+            const accountId = _extractAccountIdFromRole(targetRole);
+            const roleName = _extractRoleNameFromRole(targetRole);
+            this.clientManager = new ClientManager({
```

Comment:
should the clientManager not be created just once, in `QueueProcessor` ?

`_setupDestClients` is called multiple times in the sequence of `ReplicateObject` task, so this will repeatedly create `ClientManager`s, instead of just (re)creating (and possibly reusing) clients...

Reply:
There are two sides to this :
- `ClientManager` is made in a way that doesn't support having clients with different credentials, it only initializes the STS credentials once at init. I created this ticket BB-663 for improving that to allow sharing the same client between multiple `ReplicateObject` tasks for the same destination.
- The multiple calls to `_setupDestClients` are expected to recreate a client each time, those multiple calls are done in a failure case where we switch the host and retry again.